WorkspaceServer.Tests/Kernel/LanguageKernelTests.cs

```diff
@@ -0,0 +1,1117 @@
+
+            KernelEvents.ValuesOnly()
+                        .Should()
+                        .ContainSingle(e => e is ReturnValueProduced);
+
+            KernelEvents.ValuesOnly()
+                        .OfType<ReturnValueProduced>()
+                        .Single()
+                        .Value
+                        .Should()
+                        .Be(new { value = "hello" }.ToJson());
+        }
+
+        [Theory]
+        [InlineData(Language.CSharp)]
+        [InlineData(Language.FSharp)]
+        public async Task it_can_load_assembly_references_using_r_directive_seperate_submissions(Language language)
+        {
+            var kernel = CreateKernel(language);
+
+            // F# strings treat \ as an escape character.  So do C# strings, except #r in C# is special, and doesn't.  F# usually uses @ strings for paths @"c:\temp\...."
+            var dllPath = new FileInfo(typeof(JsonConvert).Assembly.Location).FullName.Replace('\\', '/');
+
+            var source = language switch
+            {
+                Language.FSharp => new[] {
+$"#r \"{dllPath}\"",
+"open Newtonsoft.Json",
+@"let json = JsonConvert.SerializeObject( struct {| value = ""hello"" |} )",
+"json"
+},
+
+                Language.CSharp => new[] {
+$"#r \"{dllPath}\"",
+"using Newtonsoft.Json;",
+@"var json = JsonConvert.SerializeObject(new { value = ""hello"" });",
+"json"
+}
+            };
+
+            await SubmitCode(kernel, source);
+
+            KernelEvents.ValuesOnly()
+                        .Should()
+                        .ContainSingle(e => e is ReturnValueProduced);
+
+            KernelEvents.ValuesOnly()
+                        .OfType<ReturnValueProduced>()
+                        .Single()
+                        .Value
+                        .Should()
+                        .Be(new { value = "hello" }.ToJson());
+        }
+
+        [Theory(Skip = "CommandBuilder does not support directives with quoted paths")]
+        [InlineData(Language.CSharp)]             // Todo: Consider allowinf C# #r work with @"" strings
+        [InlineData(Language.FSharp)]             // Todo: make CommandLineParser deal with @quoted strings so that F# strings work 
+        public async Task it_can_load_assembly_references_using_r_directive_at_quotedpaths(Language language)
+        {
+            var kernel = CreateKernel(language);
+
+            // F# strings treat \ as an escape character.  So do C# strings, except #r in C# is special, and doesn't.  F# usually uses @ strings for paths @"c:\temp\...."
+            var dllPath = new FileInfo(typeof(JsonConvert).Assembly.Location).FullName;
+
+            var source = language switch
+            {
+                Language.FSharp => new[] {
+$"#r @\"{dllPath}\"",
+@"
+open Newtonsoft.Json
+let json = JsonConvert.SerializeObject( struct {| value = ""hello"" |} )
+json
+"},
+                Language.CSharp => new[] {
+$"#r @\"{dllPath}\"",
```

Comment:
>$"#r @\"{dllPath}\"" [](start = 0, length = 20)

I believe this works if you remove the `@`. The issue is not the quoted path.

Reply:
Ahh … nice.